rpc/grpc/server/src/connection.rs

```diff
@@ -160,354 +226,42 @@ impl Connection {
-                }
+    pub fn get_or_register_listener_id(&self) -> ListenerId {
+        *self
+            .inner
+            .mutable_state
+            .lock()
+            .listener_id
+            .get_or_insert_with(|| self.inner.server_context.notifier.as_ref().register_new_listener(self.clone()))
+    }
 
-                Payload::StopNotifyingPruningPointUtxoSetOverrideRequest(ref request) => {
-                    let notify_request = NotifyPruningPointUtxoSetOverrideRequestMessage::from(request);
-                    let response: StopNotifyingPruningPointUtxoSetOverrideResponseMessage =
-                        match kaspa_rpc_core::NotifyPruningPointUtxoSetOverrideRequest::try_from(&notify_request) {
-                            Ok(request) => {
-                                let result = notifier
-                                    .clone()
-                                    .execute_subscribe_command(
-                                        listener_id,
-                                        Scope::PruningPointUtxoSetOverride(PruningPointUtxoSetOverrideScope::default()),
-                                        request.command,
-                                    )
-                                    .await;
-                                NotifyPruningPointUtxoSetOverrideResponseMessage::from(result).into()
-                            }
-                            Err(err) => NotifyPruningPointUtxoSetOverrideResponseMessage::from(err).into(),
-                        };
-                    KaspadResponse {
-                        id: 0,
-                        payload: Some(kaspad_response::Payload::StopNotifyingPruningPointUtxoSetOverrideResponse(response)),
-                    }
-                }
+    fn unregister_listener(&self) {
+        let listener_id = self.inner.mutable_state.lock().listener_id.take();
+        if let Some(listener_id) = listener_id {
+            self.inner.server_context.notifier.unregister_listener(listener_id).expect("unregister listener")
+        }
+    }
 
-                _ => {
-                    return Err(GrpcServerError::InvalidSubscriptionPayload);
-                }
-            }
-        } else {
-            return Err(GrpcServerError::InvalidSubscriptionPayload);
-        };
-        response.id = request.id;
+    pub fn request_channel_size() -> usize {
+        256
+    }
 
-        Ok(response)
+    /// Enqueues a response to be sent to the client
+    pub async fn enqueue(&self, response: KaspadResponse) -> GrpcServerResult<()> {
+        assert!(response.payload.is_some(), "Kaspad gRPC message should always have a value");
+        match self.inner.outgoing_route.try_send(response) {
+            Ok(_) => Ok(()),
+            Err(TrySendError::Closed(_)) => Err(GrpcServerError::ConnectionClosed),
+            Err(TrySendError::Full(_)) => Err(GrpcServerError::OutgoingRouteCapacityReached(self.to_string())),
```

Comment:
why not `await` if channel is full? (meaning use `.outgoing_route.send(response).await`)
this was the previous behavior as far as I can tell

Reply:
If undelivered messages keep accumulating up the channel capacity, there might be an issue with the connection. As discussed, better increase the capacity and disconnect the client when reaching full capacity.